source/CargoSpace/Assets/Scripts/TargetableBehavior.cs

```diff
@@ -35,26 +46,43 @@ void Update()
         
     }
 
-    
-
-    public void ApplyDamage(float strength)
+    public void OnDamaged(float strength)
     {
         if (Indestructable)
         {
             return;
         }
 
+        var previousHealth = CurrentHealth;
         CurrentHealth = Math.Max(0, CurrentHealth - strength);
 
-        if (CurrentHealth == 0)
+        const float epsilon = 0.001f;
+        if (!previousHealth.HasChanged(CurrentHealth, epsilon))
```

Comment:
This isn't a case we want in the game, we should log so we know there's an issue